### ISSUE TITLE
BitmapFonts no longer getting processed by the Loader in 5.4.0-rc.2

### ISSUE THREAD

**Issue by Article19** [pixijs/pixi.js#7008]
My Bitmap Fonts no longer seem to be recognized after being loaded. I can see the .fnt file in Loader.resources, but PIXI.BitmapFont.available returns an empty object. This worked in 5.3.3.

**Comment by bigtimebuddy** [comment]
Can you please post the font file you’re trying to load?

**Comment by Article19** [comment]
All of my bitmap fonts are made with bmGlyph. Attached is a sample.
[AvenirLtStdMd.zip](https://github.com/pixijs/pixi.js/files/5555143/AvenirLtStdMd.zip)

**Comment by bigtimebuddy** [comment]
The problem is this RegExp:

https://github.com/pixijs/pixi.js/blob/dev/packages/text-bitmap/src/formats/XMLStringFormat.ts#L22

It is expecting `<info face=` and in your example it's

```xml
<info aa='1' size='20' smooth='1' stretchH='100' bold='0' padding='0,0,0,0' spacing='0,0' charset='' italic='0' unicode='0' face='AvenirLtStdMd'/>
```

If you make `face` the first argument, that loads the font correctly. We need a much better XML check here. Should be an easy fix though.

**Comment by bigtimebuddy** [comment]
My preference would probably to attempt to parse the XML (obvi bail if unsuccessful) and check for children rather than testing with RegExp. I can be difficult to get the RegExp right and different editors generate different syntax. Also, a unit-test for this case would be important in whatever PR is submitted.

**Comment by Article19** [comment]
Can you confirm that something changed in 5.4 to break this feature? I've been using the same font files since v5 without issue, and I believe the priority should be to avoid breaking existing functionality - especially for live content. I ask in case no PR is forthcoming because perhaps a reversion to the working state should be considered as a stop-gap.

**Comment by Article19** [comment]
BTW, if the only goal of that code is to check if data is a string and contains both '<info' and 'face=' then wouldn't this get it done without having to implement an XML parser?

static test(data: unknown): boolean
{
	if (data === 'string') {
		if (data.indexOf('<info') > -1) {
			if (data.indexOf('face=') > -1) {
				return true;
			}
		}
	}
	return false;
}

**Comment by bigtimebuddy** [comment]
Yes, this was a regression. But @SerG-Y fix is good and will be sufficient to fix before 5.4.0 release. Previously, all FNT files were assumed to be XML, but that is not the case using bmGlyph

**Comment by Article19** [comment]
Sounds great.

BTW, wrt to bmGlyph, there are many export options. I use Sparrow/Starling - which does produce valid xml.

Cheers!
